### DISCUSSION TITLE
Build from source issues

### DISCUSSION THREAD

**Discussion by wheatman** [OpenCilk/opencilk-project#100] [Q&A] [answered]
I tried to build from source as opposed to using one of the prebuilt versions and ran into an issue.
When I try and compile with cilk I get the following message
```
clang-12: error: Cannot find OpenCilk runtime ABI bitcode file: libopencilk-abi.bc
```

The version I am using is 
```
../opencilk-project/build/bin/clang++ --version
clang version 12.0.0 (https://github.com/OpenCilk/opencilk-project.git 5d2851d7d0e689ecb3b893aa6abd12390b838c4b)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/wheatman/SSTGraph_private/../opencilk-project/build/bin
```

I followed the standard llvm build instructions and used the following cmake command
```
~/opencilk-project/build$ ~/cmake-3.21.1/bin/cmake -G "Unix Makefiles" -DLLVM_ENABLE_PROJECTS="clang;libcxx;libcxxabi;compiler-rt" -DCMAKE_BUILD_TYPE=RelWithDebInfo ../llvm
``` 
and then compiled with `make` which ran to completion and created the binaries

I assume I just missed something in the build process, but am not sure what it is, so if there is any instruction on building from source I can give them a try.

On this machine I also have old versions which I downloaded from the scripts of both OpenCilk-10.0.1-Linux and OpenCilk-9.0.1-Linux and I am not sure if they could be conflicting with anything.

**Comment by neboat** [comment, marked as answer]
There are a couple differences between OpenCilk's build-from-source process and LLVM's.
1. In addition to cloning the opencilk-project repo, you also need to clone cheetah (https://github.com/OpenCilk/cheetah) into `opencilk-project/cheetah` and productivity-tools (https://github.com/OpenCilk/productivity-tools) into `opencilk-project/cilktools`.
2. When you configure CMake (i.e., where you specify `-DCMAKE_BUILD_TYPE` and so on), you need to also specify `-DLLVM_ENABLE_RUNTIMES="cheetah;cilktools"`.
Aside from those two differences, the rest of the build-from-source process should be similar between OpenCilk and LLVM.

Let me know if those two changes work for you.

**Comment by wheatman** [comment]
Thanks, that worked
